conans/client/conf/__init__.py

```diff
@@ -121,7 +121,8 @@
         libcxx: [libstdc++, libc++]
         cppstd: [null, 98, gnu98, 11, gnu11, 14, gnu14, 17, gnu17, 20, gnu20, 23, gnu23]
     intel-cc:
-        version: ["2021.1", "2021.2", "2021.3"]
+        version: ["2021.1", "2021.2", "2021.3", "2021.4", "2022.1", "2022.2",
```

Comment:
Do you know what is the default binary compatibility of compiler versions? 
In ``gcc`` we define both ``6`` (the detected default) and then ``6.1``, ``6.2``, etc, including the minor, for users that want to opt-in to use the minor too, but in general the major only is good enough for general binary compatibility.

Reply:
I am newbie and therefore probably missing this point. Do you say that probably in the file it is enough to define the major version of the compiler? Meanwhile I am trying to sort out the matter of binary compatibility.

Reply:
The way the default settings.yml + auto-detection model binary compatibility for ``gcc`` is:

- The profile detection will return ``compiler=gcc`` and ``compiler.version=6`` (Just the major, excluding the minor), for all minor versions of gcc, 6.0, 6.1, 6.2, etc
- That means that the same binary compiled with any of the 6.X minors will be considered compatible for the other versions. A binary built with 6.1, will still have ``compiler.version=6`` in the profile, and result in an identical ``package_id`` than if building with 6.2, because that will still be modeled by the same ``compiler.version=6``. That is the minor is dropped from the binary model
- The minor versions in the ``settings.yml`` after the ``6``, like ``6, 6.0, 6.1, 6.2...`` are for users that want to explicitly model binary incompatibility among different gcc6 minors. They just need to provide the version explicitly in their profiles with ``compiler.version=6.2`` and that will guarantee a different binary than if writing ``compiler.version=6.1``. The tradeoff is more strict binary compatibility guarantees, at the expense of having to build from source more often, for every minor release of the compiler.

The Conan proposed "default" binary compatibility is just a balanced one, achieving binary compatibility/incompatibility in the vast majority of cases, without needed to build too much from source. 

Please let me know if this clarifies a bit.

Reply:
It's clear, thank you for taking the time to explain everything in detail. I totally agree with the "default" binary compatibility proposed in Conan. I found in the Intel website the logic behind the [versioning schema](https://www.intel.com/content/www/us/en/developer/articles/tool/oneapi-toolkit-and-component-versioning-schema.html). Intel is following the Semantic Versioning (SemVer).
A SemVer number looks like this: MAJOR.MINOR.PATCH. 

- The MAJOR release has many new features and changes when ABI/API compatibility is broken.
- The MINOR release is an update to the MAJOR release. Features can include an expanded API or new features that are backwards-compatible. 
- Intel oneAPI components may issue a critical PATCH for specific bug and security fixes.

So we could say that also in this case it makes sense to return the major release. I will update the code of method `default_compiler_version` accordingly. Thanks.